src/multitaper.jl

```diff
@@ -0,0 +1,518 @@
+        demeaned_signal = nothing
+    end
+    
+    window = dpss(n_samples, nw, ntapers)
+    evals = dpsseig(window, nw)
+
+    if low_bias
+        keep_evals_mask = evals .> 0.9
+        window, evals = window[:, keep_evals_mask], evals[keep_evals_mask]
+    end
+    ntapers = size(window, 2)
+
+    denom = fs * sum(evals) / 2
+    weighted_evals = evals ./ denom
+
+
+    mt_config = MTConfig{T}(n_samples; fs=fs, window=window, ntapers=ntapers, nw=nw, kwargs...)
+
+    x_mt = Array{fftouttype(T), 3}(undef, mt_config.n_freq_bins, mt_config.ntapers, n_channels)
+    if freq_range !== nothing
+        freq_mask = first(freq_range) .< mt_config.freq .< last(freq_range)
+        freq_inds = findall(freq_mask)
+        freq = mt_config.freq[freq_mask]
+    else
+        freq_inds = eachindex(mt_config.freq)
+        freq = mt_config.freq
+    end
+    return MTCrossSpectraConfig{T, typeof(weighted_evals), typeof(x_mt), typeof(mean_per_channel), typeof(demeaned_signal), typeof(freq), typeof(freq_range), typeof(freq_inds), typeof(mt_config)}(n_channels, weighted_evals, x_mt, demean, mean_per_channel, demeaned_signal, freq, freq_range, freq_inds,  mt_config)
+end
+
+allocate_output(config::MTCrossSpectraConfig{T}) where {T} = Array{fftouttype(T), 3}(undef, config.n_channels, config.n_channels, length(config.freq))
+
+"""
+    mt_cross_spectral!(output, signal::AbstractMatrix{T}, config::MTCrossSpectraConfig{T}) where {T} -> CrossSpectral
+
+Computes a multitapered cross spectral matrix.
+
+* `output`: `n_channels` x `n_channels` x `length(config.freq)`. Can be created by `allocate_output(config)`.
+* `signal`: `n_channels` x `n_samples`
+* `config`: `MTCrossSpectraConfig{T}`
+
+"""
+@views function mt_cross_spectral!(output, signal::AbstractMatrix{T}, config::MTCrossSpectraConfig{T}) where {T}
+    if size(signal) != (config.n_channels, config.mt_config.n_samples)
+        throw(DimensionMismatch("Size of `signal` does not match `(config.n_channels, config.mt_config.n_samples)`;
+        got `size(signal)`=$(size(signal)) but `(config.n_channels, config.mt_config.n_samples)`=$((config.n_channels, config.mt_config.n_samples))"))
+    end
+    if size(output) != (config.n_channels, config.n_channels, length(config.freq_inds))
+        throw(DimensionMismatch("Size of `output` does not match `(config.n_channels, config.n_channels, length(config.freq_inds))`;
+        got `size(output)`=$(size(output)) but `(config.n_channels, config.n_channels, length(config.freq_inds))`=$((config.n_channels, config.n_channels, length(config.freq_inds)))"))
+    end
+
+    if config.demean
+        mean!(config.mean_per_channel, signal)
+        config.demeaned_signal .= signal .- config.mean_per_channel
+        signal = config.demeaned_signal
+    end
+
+    x_mt = config.x_mt
+
+    for k in 1:config.n_channels
+        tapered_spectra!(x_mt[:,:,k], signal[k, :], config.mt_config)
+    end
+    x_mt[1, :, :] ./= sqrt(2)
+    if iseven(size(signal, 2))
```

Comment:
I kind of wonder if this should be `nfft` instead of `size(signal, 2)`, but MNE does it this way and I don't fully understand what it's for 🤷

Reply:
Seems to be an MNE bug: https://github.com/mne-tools/mne-python/issues/9350

Reply:
I've updated the code here to use `nfft` instead